src/Symfony/Component/HttpFoundation/Tests/ParameterBagTest.php

```diff
@@ -111,34 +114,154 @@ public function testHas()
-        $this->assertEquals('012', $bag->getDigits('word'), '->getDigits() gets only digits as string');
-        $this->assertEquals('', $bag->getDigits('unknown'), '->getDigits() returns empty string if a parameter is not defined');
+        $bag->getDigits('word');
     }
 
+    /**
+     * @group legacy
+     */
     public function testGetInt()
     {
-        $bag = new ParameterBag(['digits' => '0123']);
+        $this->expectDeprecation('Since symfony/http-foundation 6.3: Method "getInt" is deprecated and replaced by "getInteger".');
 
-        $this->assertEquals(123, $bag->getInt('digits'), '->getInt() gets a value of parameter as integer');
-        $this->assertEquals(0, $bag->getInt('unknown'), '->getInt() returns zero if a parameter is not defined');
+        $bag = new ParameterBag(['digits' => '123', 'bool' => true, 'string' => 'abc']);
+
+        $this->assertSame(123, $bag->getInt('digits'), '->getInt() gets a value of parameter as integer');
+        $this->assertSame(0, $bag->getInt('unknown'), '->getInt() returns zero if a parameter is not defined');
+        $this->assertSame(10, $bag->getInt('unknown', 10), '->getInt() returns the default if a parameter is not defined');
+        $this->assertSame(1, $bag->getInt('bool', 10), '->getInt() returns zero if a parameter is invalid');
+        $this->assertSame(0, $bag->getInt('string', 10), '->getInt() returns zero if a parameter is invalid');
+    }
+
+    public function testGetInteger()
+    {
+        $bag = new ParameterBag(['digits' => '123', 'bool' => true]);
+
+        $this->assertSame(123, $bag->getInteger('digits'), '->getInteger() gets a value of parameter as integer');
+        $this->assertSame(0, $bag->getInteger('unknown'), '->getInteger() returns zero if a parameter is not defined');
+        $this->assertSame(10, $bag->getInteger('unknown', 10), '->getInteger() returns the default if a parameter is not defined');
+        $this->assertSame(1, $bag->getInteger('bool'), '->getInteger() returns 1 if a parameter is true');
+    }
+
+    public function testGetIntegerExceptionWithArray()
+    {
+        $bag = new ParameterBag(['digits' => ['123']]);
+
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionMessage('Parameter value "digits" cannot be converted to "int".');
+
+        $bag->getInteger('digits');
+    }
+
+    public function testGetIntegerExceptionWithObject()
+    {
+        $bag = new ParameterBag(['object' => $this]);
+
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionMessage('Parameter value "object" cannot be filtered.');
+
+        $bag->getInteger('object');
+    }
+
+    public function testGetIntegerExceptionWithInvalid()
+    {
+        $bag = new ParameterBag(['word' => 'foo_BAR_012']);
+
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionMessage('Parameter value "word" cannot be converted to "int".');
```

Comment:
This is the only difference: `getInt` returns `0` (not the default) and `getInteger` would throw and exception.

Reply:
Instead of creating a new method. We can add a 3rd argument to the function `getInt($key, $default=0, $errorOnInvalid=false)` with a depreciation notice when the argument is not set to `true`.